src/Mod/PartDesign/Gui/TaskDressUpParameters.cpp

```diff
@@ -140,6 +140,39 @@ void TaskDressUpParameters::referenceSelected(const Gui::SelectionChanges& msg,
 
 void TaskDressUpParameters::addAllEdges(QListWidget* widget)
 {
+#ifdef FC_USE_TNP_FIX
+    if (!DressUpView) {
+        return;
+    }
+
+    PartDesign::DressUp* pcDressUp = static_cast<PartDesign::DressUp*>(DressUpView->getObject());
+    App::DocumentObject* base = pcDressUp->Base.getValue();
+    if (!base) {
+        return;
+    }
+    int count = Part::Feature::getTopoShape(base).countSubShapes(TopAbs_EDGE);
+    auto subValues = pcDressUp->Base.getSubValues(false);
+    std::size_t len = subValues.size();
+    std::string name("Edge");
+    for (int i = 0; i < count; ++i) {
+        name.resize(4);
+        name += std::to_string(i + 1);

```

Comment:
Out of curiosity - why not use string stream like it is done in the non-tnp fix code?

```
    for (int i = 0; i < count; i++){
        std::ostringstream ss;
        ss << "Edge" << ii+1;
        
        auto name = ss.str();
```

Or at least

```
std::string name = "Edge" + std::to_string(i+1);
```

Current solution with reusing the same variable, resizing it etc is very hard to follow and understand. 

Of course it would be even better to use something like `std::format` (if only we could...)

Reply:
Good point.  Because I was focused on finding the right missing code from the LS3 side to bring over, and not on the modernization of same.